src/deque.jl

```diff
@@ -188,7 +188,7 @@ julia> setindex(@SMatrix[2 4; 6 8], 1, 2)
     return quote
         @_propagate_inbounds_meta
         @boundscheck if (index < 1 || index > $(L))
-            throw(BoundsError(a, index))
+            Base.throw_boundserror(a, index)
```

Comment:
This is identical within a `@noinline`. Can we define the helper function in this package, instead of relying on `Base` internals?